rpcserver.go

```diff
@@ -1357,6 +1362,114 @@ func (r *rpcServer) ListTransfers(ctx context.Context,
 	return resp, nil
 }
 
+// SetPendingTransferProofCourierAddr sets the transfer output proof courier
+// address for a pending (undelivered proof) transfer. The anchoring transaction
+// of the target transfer must have already been confirmed on-chain.
+func (r *rpcServer) SetPendingTransferProofCourierAddr(ctx context.Context,
+	req *updateCourierAddrReq) (*updateCourierAddrResp, error) {
+
+	// Unmarshal the anchor tx hash.
+	if len(req.AnchorTxid) == 0 {
+		return nil, fmt.Errorf("anchor txid must be set")
+	}
+
+	anchorTxHash, err := chainhash.NewHashFromStr(req.AnchorTxid)
+	if err != nil {
+		return nil, fmt.Errorf("invalid anchor tx hash: %w", err)
+	}
+
+	// Validate the requested new proof courier address.
+	_, err = proof.ParseCourierAddress(req.NewProofCourierAddr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid proof courier address: %w", err)
+	}
+
+	// Unmarshal the transfer output script key public key.
+	outputScriptKey, err := parseUserKey(req.TransferOutputScriptPubKey)
```

Comment:
So the pubkey is a safety net to make sure the user really specified the right one?